packages/design-system/src/components/web-components/ds-choice/ds-choice.test.tsx

```diff
@@ -0,0 +1,274 @@
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it('applies inverse className to input', () => {
+    renderChoice({ inversed: true });
+    const input = screen.getByRole('checkbox');
+    expect(input).toHaveClass('ds-c-choice--inverse');
+  });
+
+  it('applies small className to input', () => {
+    renderChoice({ size: 'small' });
+    const input = screen.getByRole('checkbox');
+    expect(input).toHaveClass('ds-c-choice--small');
+  });
+
+  it('applies additional classNames to root element', () => {
+    const { container } = renderChoice({ className: 'foo' });
+    const el = container.firstChild as HTMLElement;
+    expect(el).toHaveClass('foo');
+  });
+
+  it('accepts a string value in the input', () => {
+    renderChoice({ value: 'bar' });
+    const input = screen.getByRole('checkbox');
+    expect(input).toHaveAttribute('value', 'bar');
+  });
+
+  it('accepts a number value in the input', () => {
+    renderChoice({ value: 100 });
+    const input = screen.getByRole('checkbox');
+    expect(input).toHaveAttribute('value', '100');
+  });
+
+  it('accepts a custom id', () => {
+    renderChoice({ 'root-id': 'custom_id' });
+    const el = screen.getByRole('checkbox');
+    expect(el.id).toBe('custom_id');
+  });
+
+  it('generates a unique id', () => {
+    const props = {
+      name: 'presidents',
+      label: defaultAttrs.label,
+      type: defaultAttrs.type,
+    };
+    const { container } = render(
+      <div>
+        <ds-choice value="a" {...props} />
+        <ds-choice value="b" {...props} />
+      </div>
+    );
+
+    const idRegex = /choice--\d+/;
+    const labels = container.querySelectorAll('label');
+    const labelA = labels[0];
+    const labelB = labels[1];
+    const inputA = screen.getByDisplayValue('a');
+    const inputB = screen.getByDisplayValue('b');
+
+    // IDs should be unique!
+    expect(inputA.id).not.toBe(inputB.id);
+
+    // First Choice
+    expect(inputA.id).toMatch(idRegex);
+    expect(labelA.getAttribute('for')).toBe(inputA.id);
+
+    // Second choice
+    expect(inputB.id).toMatch(idRegex);
+    expect(labelB.getAttribute('for')).toBe(inputB.id);
+  });
+
+  describe('event handlers and emitters', () => {
+    let props;
+
+    beforeEach(() => {
+      props = {
+        onBlur: jest.fn(),
+        onChange: jest.fn(),
+      };
+    });
+
+    it('calls the onChange handler', () => {
+      renderChoice(props);
+      const el = screen.getByRole('checkbox');
+      userEvent.click(el);
+
+      expect(props.onBlur).toHaveBeenCalledTimes(0);
+      expect(props.onChange).toHaveBeenCalledTimes(1);
+    });
+
+    it('calls the onBlur handler', () => {
+      renderChoice(props);
+      const el = screen.getByLabelText('George Washington');
+      el.focus();
+      userEvent.tab();
+
+      expect(props.onBlur).toHaveBeenCalledTimes(1);
+      expect(props.onChange).toHaveBeenCalledTimes(0);
+    });
+  });
```

Comment:
This needs to be updated to use `addEventListener` and the custom events. I think these are only being passed down successfully by a trick of Preact in the unit tests.